smarty/sampler.py

```diff
@@ -129,41 +144,16 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
                 self.atomtypes = [(smarts, typename)] + self.atomtypes
```

Comment:
Yes, @camizanette - I think what Caitlin was proposing was to ask the user to list the base types in order from most general to least general in that file. Otherwise, as you note, there will be no way to SORT them from most general to least general since it's not possible to (a) exhaustively list all SMARTS patterns in order to determine how general each is, or (b) rank SMARTS patterns based on "generality". 

So we should put the SMARTS into this file in order based on generality, and update the docs (doc strings and README.md) to indicate this requirement.